src/main/java/hudson/plugins/git/BranchSpec.java

```diff
@@ -97,7 +97,11 @@ public boolean matches(String item, EnvVars env) {
     }
 
     private String getExpandedName(EnvVars env) {
-        return env.expand(name);
+        String expandedName = env.expand(name);
+        if (expandedName.length() == 0) {
+            return "**";
```

Comment:
No, the current default behavior is to build everything for "".
From the "Branches to build" inline help: "If left blank, all branches will be examined for changes and built."

That said, the change may indeed change behavior for some corner cases. I don't know enough about the plugin to be completely sure.